var/spack/repos/builtin/packages/upcxx/package.py

```diff
@@ -4,31 +4,15 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack import *
-import llnl.util.tty as tty
-import os
 
 
 def detect_scheduler():
-    if (os.environ.get('CROSS') is None):
-        srunbin = which('srun')
-        if srunbin is None:
-            aprunbin = which('aprun')
-            if aprunbin is None:
-                tty.warn("CROSS has not been set, however "
-                         "cannot detect scheduler.")
-                return 'none'
-            else:
-                tty.warn("CROSS has not been set, however "
-                         "aprun has been found, assuming alps scheduler.")
-                return 'alps'
-        else:
-            tty.warn("CROSS has not been set, however "
-                     "srun has been found, assuming slurm scheduler.")
-            return 'slurm'
+    if which('srun'):
+        return 'slurm'
+    elif which('aprun'):
+        return 'alps'
     else:
-        tty.warn("CROSS has been set to %s by the user."
-                 % os.environ.get('CROSS'))
-        return 'user'
+        return 'none'
```

Comment:
Minor comment, not a request: the `else` above is not needed.

Reply:
I'm on it ;-)